【課題提出】其の二 『ページ大枠のレイアウト2』再　安藤

andoazumi opened the pull request:
@tommyKazuhito 
@usukesato 

課題其の二 『ページ大枠のレイアウト2』の再チャレンジを行いましたので、お手隙の際にご確認の程よろしくお願い申し上げます。
制作時間2時間

前回SIDEBARそれぞれに別名をつけて作成したと思うので、今回は擬似クラスを使ってみたらSIDEBAR2で苦戦しました。

andoazumi committed:
[Add]其の一 『ページ大枠のレイアウト1』の作成
【html】
body内の作成

【css】
二行目
:root追加

16行目以降
作成

tommyKazuhito reviewed (commented):
@andoazumi 

すべてのデバイスでCONTENTSのウインドウの余白にFITの仕様が満たせていない気がします…
あと、チャレンジは認めますが、擬似クラスを使用した、

```
.main>.sidebar:nth-child(2) {
    height: 200px;
  }
```

は、やり方としてあまり良くないかなと思います。
理由としては、構造の変化（要素の増減や順番の入れ替え等）に弱いことなどがあります。
シンプルにマルチクラスでやったほうが無難かなと。

ご確認をお願いいたします。

andoazumi commented:
@tommyKazuhito 
ご確認いただき、ありがとうございます！
余白にフィットできていないところ再確認いたします。

そういう懸念点があるのですね！
先を見通す力が弱い＆判断が甘い迷うところがたくさんあるので、判断材料いただけるのありがたいです。
どの方法でやった方がいいんだろうて思って迷うこと多いので、ブラッシュアップするためにも実際に使ったもの＋この方法も考えてみたも記述していけたらと思います。

andoazumi committed:
[Update]課題1_2再チャレンジ訂正

【css】
128行目
pc版でのヘッダーの高さをCONTENTSから引くためのものを追記しました。

andoazumi commented:
@tommyKazuhito 
お疲れさまです。

余白fitでPCのところを修正いたしました。ご確認の程よろしくお願いいたします！

andoazumi committed:
[Update]課題1_2再チャレンジ訂正

【css】
135行目
pc版の最大幅の指定を追記いたしました。

andoazumi commented:
お疲れ様です。

PCサイズ時におけるCONTENTSの最大幅指定を追加いたしました。
原因は確認不足です。改善します。

tommyKazuhito commented:
@andoazumi 

確認しました。
すいません、やっぱりすべてのデバイスで、
「高さはウインドウの余白にFIT」
の仕様が満たせていない気がします…

ちょっと仕様の書き方が曖昧で申し訳なかったかもですが、
フッターとかの要素が、画面外に行ってしまっています。

一回目のチャレンジの時はきちんとできてたと思うので挙動見比べてみてください。

usukesato commented on basic-layout/no2/htdocs/css/style.css:
```diff
@@ -14,3 +28,123 @@ body {
+  }
+
+  .footer {
+    height: 60px;
```
@andoazumi 
今回はheaderも変数を使用しているのでfooterも使用した方が良いと思います。
.contentsの計算にも使用できるので。

usukesato commented on basic-layout/no2/htdocs/css/style.css:
```diff
@@ -14,3 +28,123 @@ body {
+  }
+}
+
+@media screen and (min-width:768px) {
```
@andoazumi cc @tommyKazuhito 
PCの時にタブレットのデザインも継承できて便利ですが、完全に分けることもできます。
※最初から入っている指定のものであればこのコメントは無視してください。

`@media screen and (min-width: 768px) and (max-width: 1023px) {...`

tommyKazuhito replied:
@usukesato 
ブレークポイントは指定していますが、MQの書き方は自由にしています！

usukesato replied:
@tommyKazuhito 
ありがとうございます！

@andoazumi 
どちらの書き方もできるので書きやすい方で大丈夫ですが、
分けた方がCSSの打ち消し（上書き）とかしなくていいので分かりやすいかもしれないです。

andoazumi replied:
@tommyKazuhito 
ありがとうございます！
次ので分ける方でやってみます。

usukesato commented on basic-layout/no2/htdocs/css/style.css:
```diff
@@ -14,3 +28,123 @@ body {
+  }
+
+  .contents {
+    max-width: 1280px;
```
@tommyKazuhito 
`max-width`を指定すると、それ以上画面幅が伸びた時に余白ができてしまいますがデザイン上にその指定があります。
この指定は意図通りですか？

tommyKazuhito replied:
@usukesato 
これは、課題の仕様でそういう指定をしているのでOKです！

usukesato reviewed (commented):
@andoazumi 
2つコメントしましたのでご確認ください。

@tommyKazuhito 
冨原さん宛にもコメントしてしまったのですが、お手数ではありますがご確認をお願いいたします。

andoazumi commented:
@tommyKazuhito 
ご確認いただき、ありがとうございます！

やっと理解しました。確認不足と読解力不足でで申し訳ございません。
修正いたします。

andoazumi commented:
@usukesato 
ご確認いただき、ありがとうございます！

footerの件、組み方のルールを統一するということ失念しておりました。修正いたします。
レスポンシブを作る際に打ち消しで間違えることがあるのとその書き方も覚えたいので、次使ってみます。ありがとうございます！

andoazumi committed:
[Update]課題1_2再チャレンジ修正

【css】
14行目
sidebarとfooterを追加しました

72,95行目
sidebarとfooterの高さを追記しました。

129行目
footerの高さを追記しました。

76,80,109,113行目
変数使用に伴い変更いたしました。

andoazumi commented:
@tommyKazuhito 
@usukesato 

お疲れ様です。
修正いたしましたので、お手隙の際にご確認のほどよろしくお願い申し上げます。

メディアクエリを全て分けて書く方法は次回行います。
